File: easy/226. Invert Binary Tree/226. Invert Binary Tree.py
from typing import Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return
        
        if root.left or root.right:
            root = TreeNode(root.val, root.right, root.left)
        
        root = TreeNode(
            root.val, self.invertTree(root.left), self.invertTree(root.right)
        )
        
        return root

File: easy/226. Invert Binary Tree/226. Invert Binary Tree.go
package main

/**
 * Definition for a binary tree node.
 */

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}

	if root.Left != nil || root.Right != nil {
		root = &TreeNode{Val: root.Val, Right: root.Right, Left: root.Left}
	}

	root = &TreeNode{Val: root.Val, Left: invertTree(root.Left), Right: invertTree(root.Right)}
	return root
}
